src/main/java/cam72cam/immersiverailroading/thirdparty/CommonAPI.java

```diff
@@ -157,10 +159,22 @@ public FluidStack getFluid() {
     }
 
     public String getTag() {
+    	TagEvent tagEvent = new TagEvent(TagEvent.EventType.GetTag);
+    	MinecraftForge.EVENT_BUS.post(tagEvent);
+    	
+    	if (tagEvent.tag != null)
+    	{
+    		return tagEvent.tag;
+    	}
+    	
         return stock.tag;
     }
 
     public void setTag(String tag) {
+    	TagEvent tagEvent = new TagEvent(TagEvent.EventType.SetTag);
+    	tagEvent.tag = tag;
+    	MinecraftForge.EVENT_BUS.post(tagEvent);
+    	
         stock.tag = tag;
```

Comment:
Still set stock tag even if another mod receives the event.  This way, calling setTag from OpenComputers will allow the tags to persist even if the mod that has event handlers are removed.